alerting/src/main/kotlin/org/opensearch/alerting/BucketLevelMonitorRunner.kt

```diff
@@ -479,7 +480,7 @@ object BucketLevelMonitorRunner : MonitorRunner() {
                             val queryBuilder = if (input.query.query() == null) BoolQueryBuilder()
                             else QueryBuilders.boolQuery().must(source.query())
                             queryBuilder.filter(QueryBuilders.termsQuery(fieldName, bucketValues))
-                            sr.source().query(queryBuilder)
+                            sr.source().query(queryBuilder).sort("_seq_no", SortOrder.DESC)
```

Comment:
why are we sorting based on _seq_no?

there is already a range query based on **`period_end`** variable 

this seems incorrect

Reply:
we need to sort this because without `sort`, we get `random 10 docs in period of last 15 minutes` by default for a workflow running `every 1 min`. Now, the aggregation may have grouped `1000 docs`.
So, `10 out of 1000 docs generated` may not be the latest ones. We pass these `10 docs` to the delegated `doc-level monitor` which has already moved its `seq_no` past these `10 random docs` and hence do not generate an alert. 

`sort by seq_no` ensures we always get the `latest 10 docs out of the 1000 docs` considered for aggregation. Thus, when the doc-level monitor runs next time, it gets `latest 10 docs` and it goes on to geenrate an alert.